### ISSUE TITLE
WebRTC: Support publish with pure video or audio stream. How to only publish video or only publish audio.

### ISSUE THREAD

**Issue by shinoi2** [ossrs/srs#2513]
**Description**'

Please ensure that the markdown structure is maintained.

> Description of the issue encountered:
I hope to be able to push only video or only audio when using Webrtc for streaming.
SRS version: v4.0.95
SRS configuration (Config): rtc2rtmp.conf

**Reproduction:**

**Steps to reproduce the bug:**
For example, if I only have a microphone and no camera, I can choose to only transmit audio. Here is an example of only transmitting audio:
1. Comment out the camera-related code in `srs.sdk.js`.
2. Start SRS.
```js
    self.constraints = {
        audio: true,
        // video: {
        //     width: {ideal: 320, max: 576}
        // }
    };
    // self.pc.addTransceiver("video", {direction: "sendonly"});
    // self.pc.addTransceiver("video", {direction: "recvonly"});
```

**2.** At this point, WebRTC can transmit only audio and play pure audio. However, the RTMP stream cannot be played, and the third-party WebRTC (jswebrtc) cannot play it either.

**3.** If audio is commented out and only video is transmitted, WebRTC can transmit only video and play pure video. However, the RTMP stream has a large playback delay (about 6 seconds), the third-party WebRTC (jswebrtc) cannot play it, and it causes the SRS service to crash.

**Expected Behavior (Expect)**

**Can push only video or only audio normally, and can play normally.**

`TRANS_BY_GPT3`

**Comment by shinoi2** [comment]
The reason for the error is that when webrtc only pushes audio, encoding fails when encountering NACK. It should be that NACK is not handled properly.

`TRANS_BY_GPT3`

**Comment by gyf19** [comment]
I also want to know if the app's Flutter version has this bug. Thank you.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
Welcome Patch.

`TRANS_BY_GPT3`

**Comment by strangemk2** [comment]
Can you only push audio to HLS? It seems like there is no logic for segmenting and generating AAC in the code.

`TRANS_BY_GPT3`

**Comment by SuperChrisliu** [comment]
Has this problem been resolved now? Test it out. Webrtc only pushes audio, rtmp pulling is possible, but it takes about 7 seconds for rtmp to receive data.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
Already fixed by #3964. To use this feature, should upgrade to latest SRS player, allowing user to select audio or video only.
